pkg/controllers/hub/trafficmanagerprofile/controller.go

```diff
@@ -313,6 +315,37 @@ func generateAzureTrafficManagerProfile(profile *fleetnetv1beta1.TrafficManagerP
 	}
 }
 
+// buildAzureTrafficManagerProfileRequest assumes desired is always valid.
+func buildAzureTrafficManagerProfileRequest(current, desired armtrafficmanager.Profile) armtrafficmanager.Profile {
+	current.Location = desired.Location // reset the location fields
```

Comment:
Do we allow location to be changed?

Reply:
in the reality, there's no way for cx to change the location and we always create a global profile